src/session_impl.cpp

```diff
@@ -1887,6 +1864,50 @@ namespace {
 		}
 	}
 
+	void session_impl::remove_listen_sockets(std::vector<listen_endpoint_t>& eps)
+	{
```

Comment:
I could of course hoist the `session_impl` specific code into a function parameter, but I'm not convinced it's worth the extra boilerplate. This function is a pretty straightforward application of `remove_if`. The most likely place for bugs is the comparison function between `listen_socket_t` and `listen_endpoint_t`, and that is still being unit tested.